### DISCUSSION TITLE
HLS videos not downloading from Video Downloader Helper

### DISCUSSION THREAD

**Discussion by Lucas-Zxc** [aclap-dev/video-downloadhelper#1540] [Help and Q&A]
### Your question

So, this didn't happen before, but now, I'm trying to download a video from a website, it uses HLS, but the video won't download no matter what I do

My version of video-downloaderhelper and the CoApp
video-downloaderhelper: 9.1.0.41
CoApp: 2.0.19.0

what i am triyng to download: https://tokuzilla.net/watch/bakuage-sentai-boonboomger.html

### Addon & CoApp version and details

version: 9.1.0.41
target: mozilla
channel: beta
lang: pt-BR
coapp: {"found":true,"path":"C:\\Program Files\\DownloadHelper CoApp\\vdhcoapp.exe","version":"2.0.19","new_version":false}
license: {"unset":true}
platform: x86-64 win
UA: Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:133.0) Gecko/20100101 Firefox/133.0
{
    "never_show_no_incognito_msg_again": true,
    "view_options": {
        "all_tabs": false,
        "low_quality": false,
        "sort_by_status": true,
        "sort_reverse": false,
        "show_button_clean": true,
        "show_button_clean_all": false,
        "show_button_convert_local": false,
        "hide_downloaded": false
    },
    "open_count_store": 6,
    "successfull_dl": 1
}

**Comment by GrampaWildWilly** [comment]
I do not work for VDH.  I'm just another user.

I'm running Windows 11 Pro 64-bit version 24H2 build 26100.2605, Firefox 133.0.3 64-bit, licensed VDH 9.1.0.41 beta, CoApp 2.0.20.  That is pretty much the same setup as yours.

I visited your example page & got this:

![#01](https://github.com/user-attachments/assets/76c980ac-c618-408c-9f03-19c76fae6c62)

But when I tried to download that item on the VDH menu, I got this almost immediately:

![#02](https://github.com/user-attachments/assets/e48f7aba-6b8c-4f87-8652-1d9547ab34d1)

When I looked in the Firefox Developer Tools/Network Monitor, there was no HLS manifest, neither master nor stream.  I thought that was odd, seeing as VDH claimed to be recognizing an HLS video on this page.

So I thought maybe things would change once I clicked play on the video.  It turns out the video is a couple of screenloads down the page.  After I scrolled down to it, I clicked the play button, then paused it after a bit:

![#03](https://github.com/user-attachments/assets/a3025593-0032-4cc5-9deb-1ba2e4c9ba37)

The Network Monitor now had a few MP2T objects in it but still no manifest.  So I opened the VDH Hit Details on this item:

![#04](https://github.com/user-attachments/assets/35f06ba3-0a8c-4341-96e1-42f6b3695fd2)

There sure is a manifest there.  I have no idea how VDH got that manifest URL.  Must be some kind of magic that happens inside VDH to make that appear.  It's even stranger that VDH found that but it never showed up in the Network Monitor.  Anyway, I grabbed that URL & gave it to ffprobe:

[ffprobe.txt](https://github.com/user-attachments/files/18132614/ffprobe.txt)

As you can see, 403 Forbidden errors out the wazoo.  That would explain any failure to download.  I do so wish this sort of information were still displayed in VDH's error messages.  This would at least allow people to solve some, not all, but some of their own problems.  My personal take on this one is that there would be no point in clicking the `Report` button for this one.  There is nothing that can be done.

This web site does not have DRM protection in place.  But they've done what amounts to what I like to call stealth DRM.  They have placed some sort of password on the manifest & they have not made that password public.  That makes it impossible to download this content.  However, since this content is not protected by DRM, a screen recorder would successfully make a recording of this.  That's not as good as downloading, but it seems it's all that's available here.

**Reply by Lucas-Zxc** [reply]
but it's strange, because IDM can download, not in Mp4, in TS format, but it can (and before you say it, why don't you download via IDM, yes it's a possibility, but with VHD, I don't need to convert the video from .TS to .Mp4) 

is not something that needs to be fixed on the extension? the last update was in September
and as I said, this didn't happen before

**Reply by Lucas-Zxc** [reply]
I hope they find a way to solve this as soon as possible

**Reply by Mattmjs** [reply]
It has a mechanism that returns a 403 error if a download is attempted.
I also tried on yt-dlp & VLC but they also suffer a 403 error. Even the .ts chunks if you try to open them, a 403 error.
I tried manipulating the 720p playlist file using the edit and resend in the developer tools, by removing the token and resending
the url it doesn't 403 but then the playlist file is empty. If IDM can download it, I have no idea how they could do it.

**Reply by Lucas-Zxc** [reply]
> It has a mechanism that returns a 403 error if a download is attempted. I also tried on yt-dlp & VLC but they also suffer a 403 error. Even the .ts chunks if you try to open them, a 403 error. I tried manipulating the 720p playlist file using the edit and resend in the developer tools, by removing the token and resending the url it doesn't 403 but then the playlist file is empty. If IDM can download it, I have no idea how they could do it.

what do you mean yt-dlp, isn't it just for downloading stuff from Youtube?

**Reply by Mattmjs** [reply]
No, not only YouTube but loads of other sites it can be used for.

**Reply by GrampaWildWilly** [reply]
I have never used IDM so I don't know how it works.  But I can make some guesses.  As I stated above, once you start playing the video, a bunch of MP2T objects start showing up in the Network Monitor.  I would imagine IDM has a way of detecting & downloading those.  It assembles them into a complete video.  But as you have observed, a collection of MP2T objects welded together becomes a .TS object, not MP4, MKV, or any of the other common video file formats.  Between Matt & myself, we found no way to download anything from that site because of password protection.  I have no idea how IDM gets around that.  Maybe there's no passwords on the MP2T objects.  This is all just guessing.

**Comment by Lucas-Zxc** [comment]
by the way, do you know who develops or take care of VHD?

**Reply by GrampaWildWilly** [reply]
There is exactly one programmer working on VDH.  He is @paulrouget Paul.  He has decided it is not part of his job description to appear in this forum any more.  I disagree with this decision.  I believe part of his job is public relations, meaning he ought to make his presence felt here regularly.  But the last time I saw him in here was early August, over 4 months ago & counting.  At least @jcfrog Jérôme comes in here regularly.  Maybe on occasion he can manage to drag Paul's attention to a given problem.  I don't know how things work.  I'm like you, just another user.

**Reply by Lucas-Zxc** [reply]
> There is exactly one programmer working on VDH. He is @paulrouget Paul. He has decided it is not part of his job description to appear in this forum any more. I disagree with this decision. I believe part of his job is public relations, meaning he ought to make his presence felt here regularly. But the last time I saw him in here was early August, over 4 months ago & counting. At least @jcfrog Jérôme comes in here regularly. Maybe on occasion he can manage to drag Paul's attention to a given problem. I don't know how things work. I'm like you, just another user.

but then, how does he find out about problems or things to solve with the extension?

**Reply by GrampaWildWilly** [reply]
That's a good question.  There is the mysterious `Report` button in the error message VDH generates on some content.  That allegedly sends something actionable to somewhere that Paul looks at.  But not all problems with VDH result in that error message.  So we return to your question.  I take it as a rhetorical question.  Of course he should be coming in here & looking at problem reports.  I'm not asking.  I'm stating.  It should happen.  It does not.  I find this unacceptable.  It's not the way to run a business.

**Reply by Lucas-Zxc** [reply]
but anyway, when do you think he will solve this?

**Reply by GrampaWildWilly** [reply]
You ascribe to me powers of clairvoyance I do not possess.  I'm just another user, as I've said so many times.  I am not privvy to insider information.  I can't speculate any better than you can about when he will fix this or any other issue.  I wish he would come on here & answer these questions himself.

**Reply by Lucas-Zxc** [reply]
all right, thanks anyway

**Reply by Lucas-Zxc** [reply]
> You ascribe to me powers of clairvoyance I do not possess. I'm just another user, as I've said so many times. I am not privvy to insider information. I can't speculate any better than you can about when he will fix this or any other issue. I wish he would come on here & answer these questions himself.

and would you know of an alternative to Video Downloader Helper? for me to use at the moment

**Reply by jcfrog** [reply]
This VDH forum is not exactly the good place to promote its alternatives, thanks :)
